packages/experiments/src/components/FloatingSuggestions/Suggestions/SuggestionsStore.ts

```diff
@@ -0,0 +1,53 @@
+import { autobind } from 'office-ui-fabric-react/lib/Utilities';
+import { ISuggestionModel } from 'office-ui-fabric-react/lib/Pickers';
+
+export class SuggestionsStore<T> {
+  public suggestions: ISuggestionModel<T>[];
+
+  constructor() {
+    this.suggestions = [];
+  }
+
+  public updateSuggestions(newSuggestions: T[]): void {
+    if (newSuggestions && newSuggestions.length > 0) {
+      this.suggestions = this.convertSuggestionsToSuggestionItems(newSuggestions);
+    } else {
+      this.suggestions = [];
+    }
+  }
+
+  public getSuggestions(): ISuggestionModel<T>[] {
+    return this.suggestions;
+  }
+
+  public getSuggestionAtIndex(index: number): ISuggestionModel<T> {
+    return this.suggestions[index];
+  }
+
+  public removeSuggestion(index: number): void {
+    this.suggestions.splice(index, 1);
+  }
+
+  public convertSuggestionsToSuggestionItems(suggestions: Array<ISuggestionModel<T> | T>): ISuggestionModel<T>[] {
+    return Array.isArray(suggestions) ? suggestions.map(this._ensureSuggestionModel) : [];
+  }
+
+  @autobind
```

Comment:
Autobind is deprecated, use lambda instead. `private const _isSuggestionModel = (value...) => {func}